apps/src/applab/commands.js

```diff
@@ -1388,3 +1389,132 @@ applabCommands.getUserId = function (opts) {
   }
   return Applab.user.applabUserId;
 };
+
+/**
+ * How to execute the 'drawChartFromRecords' function.
+ * Delegates most work to ChartApi.drawChartFromRecords, but a few things are
+ * handled directly at this layer:
+ *   - Type validation (before execution)
+ *   - Queueing callbacks (after execution)
+ *   - Reporting errors and warnings (after execution)
+ * @see {ChartApi}
+ * @param {Object} opts
+ * @param {string} opts.chartId
+ * @param {ChartType} opts.chartType
+ * @param {string} opts.tableName
+ * @param {string[]} opts.columns
+ * @param {Object} opts.options
+ * @param {function} opts.callback
+ */
+applabCommands.drawChartFromRecords = function (opts) {
+  apiValidateType(opts, 'drawChartFromRecords', 'chartId', opts.chartId, 'string');
+  apiValidateType(opts, 'drawChartFromRecords', 'chartType', opts.chartType, 'string');
+  apiValidateType(opts, 'drawChartFromRecords', 'tableName', opts.tableName, 'string');
+  apiValidateType(opts, 'drawChartFromRecords', 'columns', opts.columns, 'array');
```

Comment:
I see that the spec says `columns` is required, but I also see that you added logic to infer columns when they are not specified, so I'm not quite sure what the desired behavior is here.